lib/Data/GoogleCloudStorage.php

```diff
@@ -0,0 +1,224 @@
+        }
+    }
+
+    private function _getKey($pasteid)
+    {
+        if ($this->_prefix != "") {
+            return $this->_prefix . "/" . $pasteid;
+        }
+        return $pasteid;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create($pasteid, array $paste)
+    {
+        if ($this->exists($pasteid)) {
+            return false;
+        }
+
+        try {
+            $metadata = $paste['meta'];
+            unset($metadata['attachment'], $metadata['attachmentname'], $metadata['salt']);
+
+            $this->_bucket->upload(json_encode($paste), [
+                'name' => self::_getKey($pasteid),
+                'chunkSize' => 262144,
+                'predefinedAcl' => 'private',
+                'resumable' => true,
+                'metadata' => [
+                    'content-type' => 'application/json',
+                    'metadata' => $metadata,
+                ]
+            ]);
+        } catch (\InvalidArgumentException $e) {
+            error_log("failed to upload " . $pasteid . " to " . $this->_bucket->name() . ", " .
+                trim(preg_replace('/\s\s+/', ' ', $e->getMessage())));
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function read($pasteid)
+    {
+        try {
+            $o = $this->_bucket->object(self::_getKey($pasteid));
+            $data = $o->downloadAsString();
+            return json_decode($data, true);
+        } catch (NotFoundException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete($pasteid)
+    {
+        $name = self::_getKey($pasteid);
+
+        try {
+            foreach ($this->_bucket->objects(['prefix' => $name . "/discussion/"]) as $comment) {
+                try {
+                    $this->_bucket->object($comment->name())->delete();
+                } catch (NotFoundException $e) {
+                }
+            }
+        } catch (NotFoundException $e) {
+        }
+
+        $o = $this->_bucket->object($name);
+        try {
+            $o->delete();
+        } catch (NotFoundException $e) {
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists($pasteid)
+    {
+        $o = $this->_bucket->object(self::_getKey($pasteid));
+        return $o->exists();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function createComment($pasteid, $parentid, $commentid, array $comment)
+    {
+        if ($this->existsComment($pasteid, $parentid, $commentid)) {
+            return false;
+        }
+        try {
+            $name = self::_getKey($pasteid) . "/discussion/" . $parentid . "/" . $commentid;
+            $metadata = array_key_exists('meta', $comment) ? $comment['meta'] : [];
+            unset($metadata['attachment'], $metadata['attachmentname'], $metadata['salt']);
+
+            $this->_bucket->upload(json_encode($comment), [
```

Comment:
Would it be possible to use `Json::encode()` instead of `json_encode()`? It is a very thin wrapper that detects and throws an exception when an error gets detected, by calling `json_last_error()` to detect errors of the `json_encode()` function, which otherwise might silently return a partial or even empty string.